units/systemd-timesyncd.service.in

```diff
@@ -12,7 +12,8 @@ ConditionCapability=CAP_SYS_TIME
 ConditionVirtualization=!container
 DefaultDependencies=no
 RequiresMountsFor=/var/lib/systemd/clock
-After=systemd-remount-fs.service systemd-tmpfiles-setup.service systemd-sysusers.service
+Requires=dev-rtc.device
+After=systemd-remount-fs.service systemd-tmpfiles-setup.service systemd-sysusers.service dev-rtc.device
```

Comment:
Yes, precisely. Just allocate the event source where you currently sync to the RTC, and from its handler try to sync. If the sync fails, then update the time of the event source, and restart it. If you succeed, free it. Make all that part of the Manager logic.

Also, as soon as the first adjusttimex is done, also free the event source, so that we won't bother with the RTC sync in that case anymore.